components/modal-radix.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Root,
+  Trigger as RadixTrigger,
+  Overlay,
+  Content,
+  Title,
+  Description,
+  Close,
+} from '@radix-ui/react-dialog'
+import type {
+  DialogProps,
+  DialogTriggerProps,
+  DialogOverlayProps,
+  DialogContentProps,
+  DialogTitleProps,
+  DialogDescriptionProps,
+  DialogCloseProps,
+} from '@radix-ui/react-dialog'
+import { Cross2Icon } from '@radix-ui/react-icons'
+import { PureComponent, ReactNode } from 'react'
+
+interface ModalProps extends DialogProps {
+  className?: string
+  contentClassName?: string
+  trigger: ReactNode
+  removeCloseIcon?: boolean
+  removeOverlay?: boolean
+}
+
+const DPATrigger = ({ className = '', ...rest }: DialogTriggerProps) => (
+  <RadixTrigger
+    asChild
+    className={`font-inter ${className}`.trim()}
+    {...rest}
+  />
+)
+
+const DPAOverlay = ({ className = '', ...rest }: DialogOverlayProps) => (
+  <Overlay
+    className={`fixed -inset-0 bg-black bg-opacity-50 font-inter ${className}`.trim()}
+    {...rest}
+  />
+)
+
+const StandardClose = ({ className = '', ...rest }: DialogCloseProps) => (
+  <Close {...rest} asChild>
+    <button
+      type='button'
+      className={`
+        absolute top-10px right-10px
+        flex justify-center align-middle
+        h-15px w-15px
+        font-inter
+        ${className}
+      `.trim()}
+    >
+      <Cross2Icon />
+    </button>
+  </Close>
+)
+
+const DPAContent = ({ className = '', ...rest }: DialogContentProps) => (
+  <Content
+    className={`
+      fixed top-1/2 left-1/2
+      w-90vw max-w-md max-h-85vh
+      p-7 rounded-md
+      bg-white
+      transform -translate-x-1/2 -translate-y-1/2
+      font-inter
+      ${className}
+    `.trim()}
+    {...rest}
+  />
+)
+
+const DPATitle = ({ className = '', ...rest }: DialogTitleProps) => (
+  <Title
+    className={`text-lg font-bold mb-2 font-inter ${className}`.trim()}
+    {...rest}
+  />
+)
+
+const DPADescription = ({
+  className = '',
+  ...rest
+}: DialogDescriptionProps) => (
+  <Description className={`text-gray-700 ${className}`.trim()} {...rest} />
+)
+
+const DPAClose = ({ ...rest }: DialogCloseProps) => <Close {...rest} />
+
+export class Modal extends PureComponent<ModalProps> {
```

Comment:
Exporting a pre-composed class component and exposing the sub-components as static members is an interesting approach. This is different from the "export everything as functional components and let users compose from scratch" approach we've taken with the other components. I see how providing the modal's base structure and controlling the close and veil components with props makes it simpler to use. I'm cool with this implementation, just noting that it differs a bit from the others.

Reply:
Yea, it's a pattern we use at my company when there are sub-components that also need to be exposed.  I like it because you only need to import the modal and can chain off of it, but I could also see the argument for separate functional components.  @replygirl Wouldn't mind a third opinion on whether or not we should keep this pattern.

Reply:
if you can do it without class components, i 100% support the idea of composing the partials into an opinionated default! we will need that for consistency's sake

but let's avoid class components altogether, as well as dependency hoisting like you're doing with the static members -- this is a runtime complexity increase for DX, and if anyone uses those statics, whatever they don't use from the class won't be tree-shaken out of the page on build

Reply:
i want to add unplugin-auto-import soon to make the import situation easier though. ive gotten it working with cra+craco so should be a breeze with next

Reply:
@replygirl Updated to functional components